### ISSUE TITLE
Intermittent malformed/corrupt _bulk requests from Beats

### ISSUE THREAD

**Issue by andrewkroh** [elastic/beats#35019]
Logs from Beats indicate what appears to be either malformed or corrupt `/_bulk` API requests being made by Beats to Elasticsearch. The problem is intermittent (a few per day) and happens in standalone Beats and Agent. 

Versions: Metricbeat / Filebeat 8.6.1, 8.6.2, 8.7.0

(This was [observed](https://elastic.slack.com/archives/C010LT16CFR/p1680624564329769) (internal link) by @SpencerLN.)

### Config

```yaml
    output.elasticsearch:
        hosts: 'https://${ELASTICSEARCH_ADDRESS}'
        username: ${ELASTICSEARCH_USERNAME}
        password: ${ELASTICSEARCH_PASSWORD}
        ssl:
          enabled: true
          supported_protocols: TLSv1.2
```

### Logs

Most of the errors are variations of this:

`Malformed action/metadata line [1], expected field [create], [delete], [index] or [update] but found [name]`
`Malformed action/metadata line [1], expected field [create], [delete], [index] or [update] but found [type]`
`Malformed action/metadata line [1], expected field [create], [delete], [index] or [update] but found [created]`
`Malformed action/metadata line [1], expected field [create], [delete], [index] or [update] but found [rss]`
`Malformed action/metadata line [1], expected field [create], [delete], [index] or [update] but found [stats]`
`Malformed action/metadata line [1], expected field [create], [delete], [index] or [update] but found [events]`
`Malformed action/metadata line [1], expected field [create], [delete], [index] or [update] but found [path]`
`Malformed action/metadata line [1], expected START_OBJECT but found [VALUE_STRING]`

```
{
  "@timestamp": "2023-04-01T01:31:01.903Z",
  "ecs.version": "1.6.0",
  "log.level": "error",
  "log.logger": "elasticsearch",
  "log.origin": {
    "file.line": 241,
    "file.name": "elasticsearch/client.go"
  },
  "message": "failed to perform any bulk index operations: 400 Bad Request: {\"error\":{\"root_cause\":[{\"type\":\"illegal_argument_exception\",\"reason\":\"Malformed action/metadata line [1], expected field [create], [delete], [index] or [update] but found [name]\"}],\"type\":\"illegal_argument_exception\",\"reason\":\"Malformed action/metadata line [1], expected field [create], [delete], [index] or [update] but found [name]\"},\"status\":400}",
  "service.name": "metricbeat"
}
```

But there are also a few that appear to have corrupted event data within the `_bulk` body.

`failed to perform any bulk index operations: 400 Bad Request: {"error":{"root_cause":[{"type":"x_content_parse_exception","reason":"[1:9] Unrecognized token 'elastic': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')\n at [Source: (byte[])\"elastic-`...

**Comment by cmacknz** [comment]
Have we confirmed that this isn't the result of a processor messing with the event? Can we get the configuration that was used to generate these events?

**Comment by andrewkroh** [comment]
One of the Agent instances where this was observed runs only the Slack integration. This uses no processors except for the ones added by Agent (add_host_metadata, add_cloud_metadata, add_docker_metadata, add_kubernetes_metadata).

------

What's surprising about this issue is that the bulk request action metadata line is corrupted. The bulk request body is formed like this

```json
{ "create" : { "_index" : "logs-foo.bar" } }
{ "foo" : "bar" }
{ "create" : { "_index" : "logs-foo.bar" } }
{ "foo" : "baz" }
```

and the action / metadata line should be controlled by the Elasticsearch output. The code appears to have a strict definition for the "action" values such that no arbitrary values (like "stats") can be accidentally used.

https://github.com/elastic/beats/blob/ba3bce42590dbf722061d7d92bfdb0bd903e9014/libbeat/esleg/eslegclient/bulkapi.go#L44-L57

**Comment by cmacknz** [comment]
Yes I was hoping someone would have added a processor trying to manipulate the `op_type` parameter.

Another option is that the actual event data contains invalid JSON and the parsing failure cascades into another part of the bulk request body.

**Comment by andrewkroh** [comment]
That could be possible. It would mean that whatever we are using to marshal `beat.Event`s is producing invalid JSON.

**Comment by andrewkroh** [comment]
I monitor for indexing failures our long term testing cluster. I observed this same issue in 8.9.0 coming from Filebeat running under Agent.

> failed to perform any bulk index operations: 400 Bad Request: {"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"Malformed action/metadata line [1], expected field [create], [delete], [index] or [update] but found [type]"}],"type":"illegal_argument_exception","reason":"Malformed action/metadata line [1], expected field [create], [delete], [index] or [update] but found [type]"},"status":400}

**Comment by cmacknz** [comment]
We really need to see the JSON body of the event that is failing here to get an idea of what is causing this.
